zebra-chain/src/orchard/shielded_data.rs

```diff
@@ -37,13 +37,18 @@ pub struct ShieldedData {
     pub proof: Halo2Proof,
     /// The Orchard Actions, in the order they appear in the transaction.
     /// Denoted as `vActionsOrchard` and `vSpendAuthSigsOrchard` in the spec.
-    pub actions: AtLeastOne<AuthorizedAction>,
+    pub actions: AtLeastOne<AuthorizedAction<V>>,
     /// A signature on the transaction `sighash`.
     /// Denoted as `bindingSigOrchard` in the spec.
     pub binding_sig: Signature<Binding>,
+
+    #[cfg(feature = "tx-v6")]
+    /// Assets intended for burning
+    /// TODO: FIXME: Add ref to spec
+    pub burn: V::BurnType,
 }
```

Comment:
We will appreciate the feedback this extension:
```rust
pub struct ShieldedData<V: OrchardFlavour> { ... }
```

Reply:
In general this seems like a useful design. We know it works because we used something similar for v4 and v5 transaction formats for sapling.